command/tls_cert_create.go

```diff
@@ -0,0 +1,289 @@
+			"-node":                 complete.PredictAnything,
+			"-server":               complete.PredictNothing,
+		})
+}
+
+func (c *TLSCertCreateCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+func (c *TLSCertCreateCommand) Synopsis() string {
+	return "Create a new certificate"
+}
+
+func (c *TLSCertCreateCommand) Name() string { return "tls cert create" }
+
+func (c *TLSCertCreateCommand) Run(args []string) int {
+
+	var (
+		dnsnames    flags.StringFlag
+		ipaddresses flags.StringFlag
+		ca          string
+		cli         bool
+		client      bool
+		key         string
+		days        int
+		dc          string
+		domain      string
+		node        string
+		server      bool
+	)
+
+	flags := c.Meta.FlagSet(c.Name(), FlagSetClient)
+	flags.Usage = func() { c.Ui.Output(c.Help()) }
+	flags.Var(&dnsnames, "additional-dnsname", "")
+	flags.Var(&ipaddresses, "additional-ipaddress", "")
+	flags.StringVar(&ca, "ca", "#DOMAIN#-agent-ca.pem", "")
+	flags.BoolVar(&cli, "cli", false, "")
+	flags.BoolVar(&client, "client", false, "")
+	flags.StringVar(&key, "key", "#DOMAIN#-agent-ca-key.pem", "")
+	flags.IntVar(&days, "days", 365, "")
+	flags.StringVar(&dc, "dc", "dc1", "")
+	flags.StringVar(&domain, "domain", "nomad", "")
+	flags.StringVar(&node, "node", "", "")
+	flags.BoolVar(&server, "server", false, "")
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+
+	// Check that we got no arguments
+	args = flags.Args()
+	if l := len(args); l < 0 || l > 1 {
+		c.Ui.Error("This command takes up to one argument")
+		c.Ui.Error(commandErrorText(c))
+		return 1
+	}
+	if ca == "" {
+		c.Ui.Error("Please provide the ca")
+		return 1
+	}
+	if key == "" {
+		c.Ui.Error("Please provide the key")
+		return 1
+	}
+	if !((server && !client && !cli) ||
+		(!server && client && !cli) ||
+		(!server && !client && cli)) {
+		c.Ui.Error("Please provide either -server, -client, or -cli")
+		return 1
+	}
+
+	if node != "" && !server {
+		c.Ui.Error("-node requires -server")
+		return 1
+	}
+
+	var DNSNames []string
+	var IPAddresses []net.IP
+	var extKeyUsage []x509.ExtKeyUsage
+	var name, prefix string
+
+	for _, d := range dnsnames {
+		if len(d) > 0 {
+			DNSNames = append(DNSNames, strings.TrimSpace(d))
+		}
+	}
+
+	for _, i := range ipaddresses {
+		if len(i) > 0 {
+			IPAddresses = append(IPAddresses, net.ParseIP(strings.TrimSpace(i)))
+		}
+	}
+
+	if server {
+		name = fmt.Sprintf("server.%s.%s", dc, domain)
+
+		if node != "" {
+			nodeName := fmt.Sprintf("%s.server.%s.%s", node, dc, domain)
+			DNSNames = append(DNSNames, nodeName)
+		}
+		DNSNames = append(DNSNames, name)
+		DNSNames = append(DNSNames, "localhost")
+
+		IPAddresses = append(IPAddresses, net.ParseIP("127.0.0.1"))
+		extKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth}
+		prefix = fmt.Sprintf("%s-server-%s", dc, domain)
+
+	} else if client {
+		name = fmt.Sprintf("client.%s.%s", dc, domain)
```

Comment:
Good Catch @tgross will update it accordingly

Reply:
I cant use the word -region as this is declared and used elsewhere so I was thinking that I create -cluster-region as the flag and cluster_region as the variable. 
Do you think this will be a good way forward?

Reply:
The `-region` flag only gets used for the parts of the CLI that make API calls. So you can reuse it here without needing to declare a flag; you'll need to modify the help text is all.